kt/godot-coroutine-library/src/main/kotlin/godot/coroutines/awaitResourceLoad.kt

```diff
@@ -37,51 +39,13 @@ public suspend inline fun ResourceLoader.awaitLoad(
         return this.load(path)
     }
 
-    val error = this.loadThreadedRequest(
-        path = path,
-        typeHint = typeHint,
-        useSubThreads = useSubThreads,
-        cacheMode = cacheMode,
-    )
-
-    if (error != Error.OK) {
-        GD.printErr("Could not trigger resource load. Got error: $error")
-        return null
+    // Start a new job so we have a suspension point in case the coroutine is currently in the main thread.
+    val job = GodotCoroutine.launch(Dispatchers.Default) {
+        load(path)
     }
+    job.join()
 
-    return suspendCancellableCoroutine { continuation ->
-        CoroutineScope(Dispatchers.Default).launch {
-            var success = false
-            do {
-                val progress = variantArrayOf<Any?>()
-
-                val status = awaitDeferred {
-                    this@awaitLoad.loadThreadedGetStatus(
-                        path = path,
-                        progress = progress,
-                    ).also {
-                        (progress.firstOrNull() as? RealT)?.let { onProgress(it) }
-                    }
-                }
-
-                when (status) {
-                    ResourceLoader.ThreadLoadStatus.THREAD_LOAD_LOADED -> success = true
-                    ResourceLoader.ThreadLoadStatus.THREAD_LOAD_IN_PROGRESS -> {
-                        // no op
-                    }
-
-                    ResourceLoader.ThreadLoadStatus.THREAD_LOAD_INVALID_RESOURCE,
-                    ResourceLoader.ThreadLoadStatus.THREAD_LOAD_FAILED -> continuation.resumeWithException(
-                        IllegalStateException("Failed to load resource: $status")
-                    )
-                }
-            } while (status == ResourceLoader.ThreadLoadStatus.THREAD_LOAD_IN_PROGRESS)
-
-            if (success) {
-                continuation.resume(this@awaitLoad.loadThreadedGet(path))
-            }
-        }
-    }
+    return this.load(path)
```

Comment:
As mentioned on discord, I-m not fond of this as we block a coroutine dispatcher thread altough godot has a thread pool for these kind of things. But given the fact that godot ironically does not check multiple loads for their multithreaded resource accesses I'm fine with it for now.

Reply:
Yeah it's not ideal, but the same way, using the godot pool thread is going to block a godot thread as well. And since 4.3, the pool is responsible for running servers. I think blocking a godot thread is force than blocking a coroutine thread.
This is a short term solution for now, and we'll try to offer people a new dispatcher that use Godot's pool instead.